### PR TITLE
Set different 404 strings for different behaviors on classic theme

### LINKED ISSUE #15118: [IMPROVEMENT] Make separate error templates for empty product lists and search
**Current situation**
Currently, `errors/not-found.tpl` is used for 404s, empty categories, empty product listings (new products etc.) and for empty search queries.

"Sorry for inconvinience, find what you are looking for." is eeeeew wording for these situations. At first, upon 1.7 install I was wondering if my rewrites are working, I thought these are 404s.

**Proposed solution**
Make templates with different wordings like "There are no products in this category." and use it in `catalog/listing/product-list.tpl`.

- Search: "There are no products that match your criteria. Try to search something else."
- New products: "There are no new products this time."
- Empty category: "There no products in this category."
etc.

**Specifications**
https://github.com/PrestaShop/prestashop-specs/pull/200/files

**Additional information**
PrestaShop version: 1.7.6

### PR DESCRIPTION
**Pull request by NeOMakinG**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | strings for 404 were the same everywhere and it wasn't optimized
| Type?             | improvement
| Category?         | FO
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | Fixes #15118.
| How to test?      | Try different searches on FO, or see issue :)
| Possible impacts? | Search strings, no product strings

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/22908)

### CODE REVIEW

File: themes/classic/templates/catalog/listing/search.tpl
Comment (Julievrz) [inline themes/classic/templates/catalog/listing/search.tpl:8]:
```suggestion
{block name="searchSubtitle"}{l s='Please try other keywords to describe what you are looking for.' d='Shop.Theme.Global'}{/block}
```
Just a little change to match the specs. :) Thank you!

File: themes/classic/templates/errors/not-found.tpl
Comment (Julievrz) [inline themes/classic/templates/errors/not-found.tpl]: Hi! These strings ("Sorry for the inconvenience" and "Search again what you are looking for." were supposed to be canceled and replaced by the new wordings.

File: themes/classic/templates/catalog/listing/search.tpl
Comment (Julievrz) [inline themes/classic/templates/catalog/listing/search.tpl]:
```suggestion
{block name="searchSubtitle"}{l s='Please try other keywords to describe what you are looking for.' d='Shop.Theme.Global'}{/block}
```

Just a little suggestion to match with Louise's specs. Thanks!

File: themes/classic/templates/errors/not-found.tpl
Comment (NeOMakinG) [reply themes/classic/templates/errors/not-found.tpl]: Oh yes, my bad

**Review by Progi1984 — DISMISSED** [review]

**Comment by NeOMakinG** [comment]
@Julievrz are warning fine now? :)

File: themes/classic/templates/catalog/listing/search.tpl
Comment (Julievrz) [inline themes/classic/templates/catalog/listing/search.tpl]:
```suggestion
{block name="searchTitle"}{l s='No matches were found for your search' d='Shop.Theme.Catalog'}{/block}
```

File: themes/classic/templates/catalog/listing/search.tpl
Comment (Julievrz) [inline themes/classic/templates/catalog/listing/search.tpl]:
```suggestion
{block name="searchSubtitle"}{l s='Please try other keywords to describe what you are looking for.' d='Shop.Theme.Catalog}{/block}
```

File: themes/classic/templates/errors/not-found.tpl
Comment (Julievrz) [inline themes/classic/templates/errors/not-found.tpl]:
```suggestion
      <h4>{block name="searchTitle"}{l s='No matches were found for your search' d='Shop.Theme.Catalog'}{/block}</h4>
```

File: themes/classic/templates/errors/not-found.tpl
Comment (Julievrz) [inline themes/classic/templates/errors/not-found.tpl]:
```suggestion
      <p>{block name="searchSubtitle"}{l s='Please try other keywords to describe what you are looking for.' d='Shop.Theme.Catalog'}{/block}</p>
```

**Review by Julievrz — COMMENTED** [review]
Hi, I have just corrected the domains. I am going to update the specs too. 'Search our catalog' for the placeholder input should be in Shop.Theme.Global. (If I understand well, it should be possible to keep it now).

**Review by Progi1984 — DISMISSED** [review]

**Review by atomiix — DISMISSED** [review]

**Review by PierreRambaud — DISMISSED** [review]

File: themes/classic/templates/errors/not-found.tpl
Comment (Hlavtox) [inline themes/classic/templates/errors/not-found.tpl]: @NeOMakinG Can we do it like this? The block can be present only once, and renamed for better readability. 👍 

```suggestion
    {block name="notFoundTitle"}
      {if isset($title)}
        <h4>{$title}</h4>
      {else}
        <h4>{l s='No matches were found for your search' d='Shop.Theme.Catalog'}</h4>
      {/if}
    {/block}
```

Comment (NeOMakinG) [reply themes/classic/templates/errors/not-found.tpl]: We also need to modify this template then : themes/classic/templates/errors/404.tpl

Comment (Hlavtox) [reply themes/classic/templates/errors/not-found.tpl]: @NeOMakinG And what if you do it like this??

    {block name="notFoundContent"}
      {if isset($notfoundcontent)}
        {$notfoundcontent}
      {else}
        <h4>{l s='No matches were found for your search' d='Shop.Theme.Catalog'}</h4>
        <p>{l s='Please try other keywords to describe what you are looking for.' d='Shop.Theme.Catalog'}</p>
      {/if}
    {/block}

Comment (Hlavtox) [reply themes/classic/templates/errors/not-found.tpl]: Then product list and 404

```
{capture assign="notFoundContent"}
  <h4>{l s='This page could not be found' d='Shop.Theme.Global'}</h4>
  <p>{l s='Try to search our catalog, you may find what you are looking for!' d='Shop.Theme.Global'}</p>
{/capture}
```

**Comment by NeOMakinG** [comment]
@Julievrz wording domains changed and it should be fine now, looks more relevant

**Comment by Julievrz** [comment]
Hi @NeOMakinG! I had discussed it with Louise at the time and we thought that the 3 following strings should be in Shop.Theme.Global, as they are more generic and could be used in many pages of the software:

'This page could not be found.'
'Try to search our catalog, you may find what you are looking for!' 


Also, I don't see the string 'Search our catalog' in the PR

**Review by atomiix — DISMISSED** [review]

File: themes/classic/templates/catalog/listing/product-list.tpl
Comment (Hlavtox) [inline themes/classic/templates/catalog/listing/product-list.tpl]: @NeOMakinG Don't forget to change it here ;-)

File: themes/classic/templates/errors/not-found.tpl
Comment (PierreRambaud) [inline themes/classic/templates/errors/not-found.tpl]: You should keep the same wording type for block name:  `error_content`

File: themes/classic/templates/catalog/listing/search.tpl
Comment (PierreRambaud) [inline themes/classic/templates/catalog/listing/search.tpl]: It's weird to have a block here and not capture like other spots :thinking:

Comment (NeOMakinG) [reply themes/classic/templates/catalog/listing/search.tpl]: Because it extends, instead of include a file :/

**Review by atomiix — APPROVED** [review]

**Review by atomiix — APPROVED** [review]

File: themes/classic/templates/catalog/listing/search.tpl
Comment (PierreRambaud) [reply themes/classic/templates/catalog/listing/search.tpl]: trololo

**Review by PierreRambaud — APPROVED** [review]

**Review by Progi1984 — APPROVED** [review]

**Review by Julievrz — COMMENTED** [review]
I think the placeholder input in the search bar ('Search our catalog' in Shop.Theme.Catalog) is missing.

**Comment by NeOMakinG** [comment]
> I think the placeholder input in the search bar ('Search our catalog' in Admin.Global) is missing.

I can't find it, maybe @PrestaShop/prestashop-maintainers a backend developer can help to modify this placeholder ? It's specific

**Comment by Julievrz** [comment]
> > I think the placeholder input in the search bar ('Search our catalog' in Admin.Global) is missing.
> 
> I can't find it, maybe @PrestaShop/prestashop-maintainers a backend developer can help to modify this placeholder ? It's specific

Just to specify that the wording doesn't need to be changed, just added. :) It already exists in Shop.Theme.Catalog, and not Admin.Global, sorry

**Comment by Hlavtox** [comment]
@Julievrz @NeOMakinG It's ps_searchbar module's string, so search there. But I think that placeholder is outside the scope of this PR. Any hook module can hook into that place and show it's search bar. ;-)

**Comment by NeOMakinG** [comment]
@Julievrz the wording of the search bar is not blocking this PR, it should be done on a separate one if you want to add a wording on a module

**Comment by Julievrz** [comment]
Ok thanks, I'll add the wording ✔️  label and create a new issue.

**Comment by Progi1984** [comment]
Thanks @NeOMakinG & @sarahdib

**Pull request merged**
